fetch_artist_genre.py

```
@@ -0,0 +1,151 @@
+import sys
+import subprocess
+from pathlib import Path
+
+VENV_DIR = Path.home() / ".venvs" / "scripts"
+VENV_PYTHON = VENV_DIR / "bin" / "python3"
+
+# If not running inside the venv, set it up and re-launch
+if Path(sys.executable).resolve() != VENV_PYTHON.resolve():
+    if not VENV_PYTHON.exists():
+        print("Setting up virtual environment...")
+        subprocess.run([sys.executable, "-m", "venv", str(VENV_DIR)], check=True)
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)